allennlp/common/tensor.py

```diff
@@ -0,0 +1,23 @@
+from typing import Dict, Optional, Union
+
+import torch
+from torch.autograd import Variable
+import numpy
+
+DictOfArrays = Dict[str, Union['DictOfArrays', numpy.array]]  # pylint: disable=invalid-name
```

Comment:
Oops...  I had that all over the place in `DeepQA`.  Useful for documentation and readability, but clearly not correctly typed.